invenio_rdm_records/services/access/service.py

```diff
@@ -385,6 +387,14 @@ def bulk_create_grants(self, identity, id_, data, expand=False, uow=None):
             raise GrantExistsError()
 
         for grant in grants:
+            # checks if groups are enabled in the instance
```

Comment:
as we briefly chat, to improve performance, can we check it only the first time?
```
groups_enabled_checked = False

...

     if grant["subject"]["type"] == RecordAccessService.group_subject_type and not groups_enabled_checked:
         # check permissions once to improve performance
         self.require_permission(...
         groups_enabled_checked = True
```

Reply:
In the current implementation, on the first occurrence of the group grant if groups are disabled, the application fails with `PermissionDenied` error and doesn't check the rest of the grants

Reply:
shouldn't these if conditionals just be moved to `self.require_permission` instead? we would have less occurrences, and it would be checked any time permissions are checked (no need to remember to add it in case there is a new method in the service)

Reply:
on another note, just for me to understand: why do we need the conditional statements here if we have the permission generator GroupsEnabled?

Reply:
In the beginning, I did it via using permissions, but then chatted with Nico and decided that checking it with a simple `if config` would be much faster. Moreover, `require_permission` is checking the permission of the current **identity**, but whether groups are enabled or not does not depend on any identity, it depends only on the feature flag